metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java

```diff
@@ -212,29 +204,39 @@ public void report(
         final SortedMap<String, Counter> counters,
         final SortedMap<String, Histogram> histograms,
         final SortedMap<String, Meter> meters,
-        final SortedMap<String, Timer> timers) {
+        final SortedMap<String, Timer> timers
+    ) {
         final long now = System.currentTimeMillis();
 
         try {
             influxDb.flush();
 
             reportGauges(gauges, now);
 
-            for (Map.Entry<String, Counter> entry : counters.entrySet()) {
-                reportCounter(entry.getKey(), entry.getValue(), now);
-            }
+            counters.forEach((key, counter) -> {
+                parse(key, (name, tags) -> reportCounter(name, counter, now, tags));
+            });
 
-            for (Map.Entry<String, Histogram> entry : histograms.entrySet()) {
-                reportHistogram(entry.getKey(), entry.getValue(), now);
-            }
+            histograms.forEach((key, histogram) -> {
+                if (canSkipMetric(key, histogram)) {
```

Comment:
@xiaodong-izettle Would need to do `histograms.entrySet().stream.filter(...).forEach(...)` - does it worth? (not sure memory/performance wise what happens under the hood).

@fredrikbackstrom Sorry I did not think about it. If we decide to not make java8 a must I can refactor the codes. However one question there: what's the elegant way to do the `parse()` method then as I want to 2 returns from that method: name and tags.